tests/e2e/system_test/test_graceful_nodes_shutdown.py

```diff
@@ -0,0 +1,708 @@
+            amount=5,
+            mcg_obj=mcg_obj_session,
+        )
+        second_bucket_set = set(written_objects)
+        second_bucket_set.update(standard_test_obj_list)
+        assert second_bucket_set == {
+            obj.key
+            for obj in mcg_obj_session.s3_list_all_objects_in_bucket(
+                self.second_bucket_name
+            )
+        }, "Needed uploaded objects could not be found"
+        compare_bucket_object_list(
+            mcg_obj_session, self.first_bucket_name, self.second_bucket_name
+        )
+
+    def validate_data_integrity(self):
+
+        logger.info("Compare bucket replication after reboot (unidirectional)")
+        assert compare_bucket_object_list(
+            self.mcg_obj_session, self.source_bucket_name, self.target_bucket_name
+        ), f"Failed on data validation for buckets {self.source_bucket_name} and {self.target_bucket_name}"
+
+        logger.info("Compare bucket replication after reboot (bidirectional)")
+        compare_bucket_object_list(
+            self.mcg_obj_session, self.first_bucket_name, self.second_bucket_name
+        )
+
+        logger.info("Compare bucket object after reboot")
+        for idx, bucket_name in enumerate(self.bucket_names_list):
+            after_bucket_object_set = {
+                obj.key
+                for obj in self.mcg_obj.s3_list_all_objects_in_bucket(bucket_name)
+            }
+            assert (
+                self.before_bucket_object_list[idx] == after_bucket_object_set
+            ), f"Data integrity failed for S3 bucket {bucket_name}"
+
+        logger.info("Compare PVCs MD5Sum after reboot")
+        assert self.ne_pvc_orig_md5_sum == pod.cal_md5sum(
+            self.ne_pvc_pod_obj, self.ne_file_name
+        ), (
+            f"Data integrity failed for file '{self.ne_file_name}' "
+            f"on non encrypted pvc {self.ne_pvc_obj.name} on pod {self.ne_pvc_pod_obj.name}"
+        )
+
+        assert self.eb_pvc_orig_md5_sum == pod.cal_md5sum(
+            self.eb_pvc_pod_obj, self.eb_file_name
+        ), (
+            f"Data integrity failed for file '{self.eb_file_name}' "
+            f"on encrypted block pvc {self.eb_pvc_obj.name} on pod {self.eb_pvc_pod_obj.name}"
+        )
+
+        assert self.efs_pvc_orig_md5_sum == pod.cal_md5sum(
+            self.efs_pvc_pod_obj, self.efs_file_name
+        ), (
+            f"Data integrity failed for file '{self.efs_file_name}' "
+            f"on encrypted fs pvc {self.efs_pvc_obj.name} on pod {self.efs_pvc_pod_obj.name}"
+        )
+
+    @system_test
+    @polarion_id("OCS-3976")
+    @skipif_no_kms
+    @skipif_ocs_version("<4.11")
+    def test_graceful_nodes_shutdown(
+        self, pgsql_factory_fixture, couchbase_factory_fixture, nodes
+    ):
+        """
+        Steps:
+          1) Have a cluster with FIPS, hugepages, encryption enabled
+          2) Create some resources: Create PVC (encrypt & non encrypt),
```

Comment:
https://github.com/red-hat-storage/ocs-ci/blob/master/ocs_ci/ocs/longevity_helpers.py#L204
any chance if we could use it here or improvise it to include encrypt and non encrypt ?

Reply:
I don't see any added value in using this function.
In this test, the data integrity takes place after node restart so this part is not needed.
Also, it will only spare 2 lines of code calling snapshot & restore (which I'm using an already implemented methods from conftest)